src/phoenix/experimental/evals/models/openai.py

```diff
@@ -151,11 +217,77 @@ def _build_messages(
     def _verbose_generation_info(self) -> str:
         return f"OpenAI invocation parameters: {self.public_invocation_params}"
 
-    def _generate(self, prompt: str, **kwargs: Dict[str, Any]) -> str:
+    def generate(
+        self, prompts: List[str], instruction: Optional[str] = None, num_consumers: int = 20
+    ) -> List[str]:
+        return asyncio.run(self._generate_async(prompts, instruction, num_consumers=num_consumers))
+
+    async def _generate_async(
+        self,
+        prompts: List[str],
+        instruction: Optional[str] = None,
+        num_consumers: int = 20,
+    ) -> List[str]:
+        printif(self._verbose, f"Generating responses for {len(prompts)} prompts...")
+        if extra_info := self._verbose_generation_info():
+            printif(self._verbose, extra_info)
+        if not is_list_of(prompts, str):
+            raise TypeError(
+                "Invalid type for argument `prompts`. Expected a list of strings "
+                f"but found {type(prompts)}."
+            )
+        outputs = ["unset"] * len(prompts)
+        queue: asyncio.Queue[Tuple[int, str, Optional[str]]] = asyncio.Queue(
+            maxsize=2 * num_consumers
+        )
+        SENTINEL = object()  # indicates when the queue is done
```

Comment:
Can we rename this variable more precisely?
```suggestion
        END_OF_QUEUE = object()  # sentinel indicating when the queue is done
```

Reply:
this is great